src/app-layout/visual-refresh-toolbar/toolbar/index.tsx

```diff
@@ -64,7 +65,8 @@ function convertLegacyProps(toolbarProps: ToolbarProps, legacyProps: AppLayoutIn
     globalDrawers: toolbarProps.globalDrawers ?? legacyProps.globalDrawers,
     activeGlobalDrawersIds: toolbarProps.activeGlobalDrawersIds ?? legacyProps.activeGlobalDrawersIds,
     onActiveGlobalDrawersChange: toolbarProps.onActiveGlobalDrawersChange ?? legacyProps.onActiveGlobalDrawersChange,
-    hasNavigation: toolbarProps.hasNavigation ?? !!legacyProps.navigation,
+    hasNavigation:
+      typeof toolbarProps.hasNavigation === 'boolean' ? toolbarProps.hasNavigation : !!legacyProps.navigation,
```

Comment:
this type check does not help,   somehow we need to stop legacyProps.navigation from overwriting hasNavigation when navigationHide is true.